### PR TITLE
get working

### PR DESCRIPTION
**Pull request by jquense**

### CODE REVIEW

**Comment by jquense** [comment]
I eventually want to put all the protractor wrapping and setup stuff here as well

File: src/flatten.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+export default function flatten(arr) {
```
Comment (taion) [inline src/flatten.js:1]: Is this really so deep that the standard recursive approach doesn't work?

```js
function flatten(arr, result = []) {
  for (const el of arr) {
    if (!Array.isArray(el)) {
      result.push(el);
    } else {
      flatten(el, result);
    }
  }

  return result;
}
```

File: src/framework-hook.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    if (!isKnownRoot && !isUnmounting) mountedRoots.add(root);
+    else if (isKnownRoot && isUnmounting) mountedRoots.delete(root);
+  },
+
+  onCommitFiberUnmount(rendererId, root) {
+    const currentRoots = hook.getFiberRoots(rendererId);
+    if (currentRoots.has(root)) currentRoots.delete(root);
+  },
+
+  set _relayInternals(internals) {
```
Comment (taion) [inline src/framework-hook.js:53]: magic!

File: src/flatten.js (hunk 1 above)
Comment (jquense) [reply src/flatten.js:1]: Idk it's forked from enzyme and react

Comment (taion) [reply src/flatten.js:1]: ahhh okay

Comment (jquense) [reply src/flatten.js:1]: I should probably make that clear tho

**Pull request merged**